Resources/Private/UiPlugin/src/Components/SelectFormatButtonComponent.js

```diff
@@ -0,0 +1,118 @@
+import style from './SelectFormatButton.css';
+
+export default (commandName, configuration) => {
+    @connect($transform({
+        formattingUnderCursor: selectors.UI.ContentCanvas.formattingUnderCursor
+    }))
+    @neos(globalRegistry => ({
+        i18nRegistry: globalRegistry.get('i18n')
+    }))
+    class SelectFormatButtonComponent extends PureComponent {
+        static propTypes = {
+            i18nRegistry: PropTypes.object,
+            tooltip: PropTypes.string,
+            formatOptions: PropTypes.object,
+            formattingUnderCursor: PropTypes.objectOf(PropTypes.oneOfType([
+                PropTypes.number,
+                PropTypes.bool,
+                PropTypes.string,
+                PropTypes.object
+            ])),
+        };
+
+        constructor(props) {
+            super(props);
+            this.state = {
+                isOpen: false,
+                attributes: props.formattingUnderCursor[commandName],
+            };
+        }
+
+        componentWillReceiveProps = (nextProps) => {
+            // If the new selection doesn't have any formatting close the dialog
+            if (!$get(commandName, nextProps.formattingUnderCursor)) {
+                this.setState({isOpen: false});
+            } else {
+                this.setState({
+                    isOpen: this.state.isOpen,
+                    attributes: nextProps.formattingUnderCursor[commandName],
+                });
+            }
+        };
+
+        handleClick = () => {
+            this.setState({isOpen: !this.state.isOpen});
+        };
+
+        handleRemoveFormatClick = () => {
+            this.setState({
+                attributes: Object.keys(configuration.formatting.attributes).reduce((map, attribute) => {
+                    map[attribute] = '';
+                    return map;
+                }, {})
+            }, this.handleSelectionChange);
+        };
+
+        handleSelectionChange = () => {
+            executeCommand(commandName, this.state.attributes);
+        };
+
+        handleAttributeChange = (attributeName, value) => {
+            this.setState({
+                attributes: {
+                    ...this.state.attributes,
+                    [attributeName]: value,
+                }
+            }, this.handleSelectionChange);
+        };
+
+        render() {
+            const {icon, id, tooltip, i18nRegistry} = this.props;
+            const {isOpen} = this.state;
+            const {attributes} = configuration.formatting;
+            
+            return (                            
+                <div>
+                    <IconButton
+                        icon={icon}
+                        id={id}
+                        isActive={isOpen}
+                        onClick={this.handleClick}
+                        title={i18nRegistry.translate(tooltip)}
+                    />
+                    {isOpen ? (
```

Comment:
Why are you creating a group of SelectBoxes instead of a singel SelectBox? Is there a reason for that?

Reply:
This is for each Attribute (in the example above `color` and `size`). There can be multiple attributes/styles in one styling button.

Reply:
Exactly